recipe/meta.yaml

```diff
@@ -32,7 +32,8 @@ requirements:
     - protobuf
     - libprotobuf
     - pytest-runner 4.4
-    - pybind11 2.9.2
+    - pybind11 2.9.2   # [py<311]
```

Comment:
We can even remove the pinning for pybind11 as this is a python build tool.
The next linter update will not ask for pins on this one: https://github.com/anaconda-distribution/anaconda-linter/pull/188/files#diff-e63f01cdb361f795fe053fe66a8f9c5c7842c9864e2f908a09ea87d7dc66276aR45